htdocs/search.php

```diff
@@ -1108,17 +1120,11 @@
         }
 
         // additional options
-        if (!isset($options['f_userowner'])) {
-            $options['f_userowner'] = '0';
-        }
-        if ($options['f_userowner'] != 0) {  // Ocprop
+        if ($options['f_userowner'] === 1) {  // Ocprop
```

Comment:
Agree that this strict typisation is problematic; the code becomes less robust and harder to verify. What is the purpose of this fix? What is the bug to be fixed?